src/Async/index.js

```diff
@@ -48,14 +48,22 @@ function fromNode(fn, ctx) {
     throw new TypeError('Async.fromNode: CPS function required')
   }
 
-  return (...args) =>
-    Async((reject, resolve) => {
+  const _fn = curry(fn)
+
+  return (...args) => {
+
+    if (args.length < _fn.length - 1) {
+      return fromNode(_fn(...args), ctx)
+    }
+
+    return Async((reject, resolve) => {
```

Comment:
@evilsoft I know there is the issue with `Async` swallowing errors in one particular scenario, for this change, will wrapping this in the `curry` and the recursive call to `fromNode` create a similiar scenario?

Reply:
Nah, we should be good with this.
The Async "issue" is actually the desired behavior when dealing with `Promises`. And it is a behavior of Promises, so we should be good here!